internal/prober/prober_test.go

```diff
@@ -33,192 +42,159 @@ import (
 	"github.com/go-logr/logr"
 	"github.com/golang/mock/gomock"
 	. "github.com/onsi/gomega"
-
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/runtime/schema"
-	"k8s.io/apimachinery/pkg/version"
-	"k8s.io/client-go/kubernetes"
-	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/client/fake"
 )
 
 var (
-	config                              *papi.Config
-	ctrl                                *gomock.Controller
-	mds                                 *mockscaler.MockScaler
-	msc                                 *mockprober.MockShootClientCreator
-	clientBuilder                       *fake.ClientBuilder
-	fakeClient                          client.WithWatch
-	mki                                 *mockinterface.MockInterface
-	mdi                                 *mockdiscovery.MockDiscoveryInterface
-	errNotIgnorable                     = errors.New("not Ignorable error")
-	internalProbeFailureBackoffDuration = metav1.Duration{Duration: time.Millisecond}
-	proberTestLogger                    = logr.Discard()
-	defaultProbeTimeout                 = metav1.Duration{Duration: 40 * time.Second}
+	ctrl                        *gomock.Controller
+	mockScaler                  *mockscaler.MockScaler
+	mockShootClientCreator      *mockprober.MockShootClientCreator
+	mockKubernetesInterface     *mockinterface.MockInterface
+	mockDiscoveryInterface      *mockdiscovery.MockDiscoveryInterface
```

Comment:
defining these at the top might create problems if someone wishes to run these tests in parallel. I would rather create a struct containing all such mocked interfaces and pass it along after creating it for every test case.